src/ngraph/op/fused/strided_slice.cpp

```diff
@@ -0,0 +1,228 @@
+    if (begin_shape.rank().is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<size_t>(begin_shape.rank()) == 1,
+                              "Begin input must be 1D (begin rank: ",
+                              begin_shape.rank(),
+                              ").");
+    }
+    if (begin_shape[0].is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<size_t>(begin_shape[0]) == mask_size,
+                              "Begin input must have: ",
+                              mask_size,
+                              " elements, but have: ",
+                              begin_shape[0]);
+    }
+
+    const auto& end_shape = get_input_partial_shape(2);
+    if (end_shape.rank().is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<size_t>(end_shape.rank()) == 1,
+                              "End input must be 1D (end rank: ",
+                              end_shape.rank(),
+                              ").");
+    }
+    if (end_shape[0].is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<size_t>(end_shape[0]) == mask_size,
+                              "End input must have: ",
+                              mask_size,
+                              " elements, but have: ",
+                              end_shape[0]);
+    }
+}
+
+AxisSet op::StridedSlice::convert_mask_to_axis_set(const std::vector<int64_t>& mask) const
+{
+    AxisSet axis_set{};
+    for (auto i = 0; i < mask.size(); ++i)
+    {
+        if (mask[i] == 1)
+        {
+            axis_set.emplace(i);
+        }
+    }
+    return axis_set;
+}
+
+NodeVector op::StridedSlice::decompose_op() const
+{
+    Output<Node> strided_slice_result;
+    const auto stride_value_provided = get_input_size() == 4;
+    if (stride_value_provided)
+    {
+        strided_slice_result =
+            make_shared<op::DynSlice>(input_value(0),
+                                      input_value(1),
+                                      input_value(2),
+                                      input_value(3),
+                                      convert_mask_to_axis_set(m_begin_mask),
+                                      convert_mask_to_axis_set(m_end_mask),
+                                      convert_mask_to_axis_set(m_new_axis_mask),
+                                      convert_mask_to_axis_set(m_shrink_axis_mask),
+                                      convert_mask_to_axis_set(m_ellipsis_mask));
+    }
+    else
+    {
+        const auto added_axes = std::count_if(
+            m_new_axis_mask.begin(), m_new_axis_mask.end(), [](size_t i) { return i == 1; });
+        const auto shrinked_axes = std::count_if(
+            m_shrink_axis_mask.begin(), m_shrink_axis_mask.end(), [](size_t i) { return i == 1; });
+        const auto stride_size = m_begin_mask.size() + added_axes - shrinked_axes;
```

Comment:
I am sure if we can use mask size to calculate `stride` shape. The better option is probably `get_input_partial_shape(0).rank()`, but it is not available for dynamic case